components/merkledb/src/views/metadata.rs

```diff
@@ -241,6 +237,16 @@ impl<V> IndexMetadata<V> {
     pub fn index_type(&self) -> IndexType {
         self.index_type
     }
+
+    /// Returns a globally unique numeric index identifier.
+    /// MerkleDB assigns a unique numeric ID for each fully-qualified index name.
+    ///
+    /// Valid identifiers are non-zero.
```

Comment:
Shall we enforce it (in the constructor of this structure)? Or move the last two sentences to the thing that enforces these properties and link from here (@slowli , @popzxc)?

Reply:
I'd prefer the latter option.

Reply:
`IndexesPool::create_index_metadata` has a comment saying "Identifier should be non-zero to translate to a correct id in `ResolvedAddress`". `ResolveAddress::new` has a corresponding signature `(name, id: Option<NonZeroU64>)`. I'm not sure where should I put these sentences.

Reply:
A provocative idea: Is it possible to change identifier to `NonZeroU64`?

Reply:
So the `1` would be the initial value?

Reply:
I mean define `IndexMetadata.identifier` as `NonZeroU64` and return `NonZeroU64` from the corresponding public method.